torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -503,6 +527,8 @@ def __init__(
         epilogue_creator: Optional[Callable[[ir.Buffer], ir.Pointwise]] = None,
         should_block_weights: bool = True,
         name="packed_gemm",
+        is_woq_int4=False,
```

Comment:
I think `is_woq_int4` may not be a good parameter for constructing `CppGemmTemplate`. Maybe a subclass `CppWOQInt4GemmTemplate(CppGemmTemplate)` which with a class method `cls.is_woq_int4` be a better option. cc @jgong5 what do you think?

Reply:
Agreed.

Reply:
Thanks. I have refactored this part and now subclassing is used to avoid changes here.